src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt

```diff
@@ -6,14 +6,17 @@ package com.github.doyaaaaaken.kotlincsv.client
 internal class BufferedLineReader(
     private val br: Reader
 ) {
+    companion object {
+        private val BOM = '\uFEFF'
+    }
 
     fun readLineWithTerminator(): String? {
         val sb = StringBuilder()
         do {
             val c = br.read()
 
             if (c == -1) {
-                if (sb.isEmpty()) {
+                if (sb.isEmpty() || (sb.length == 1 && sb[0] == BOM)) {
```

Comment:
This implementation does not care in which line we found the BOM.
In other words, if second line contains BOM, this implementation treats the line as empty and return `null`.

I think this is enough for this class, but if necessary it is possible to add `private var isFirstLine: Boolean` to return `null` only when we read the first line.

```diff
diff --git a/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt b/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
index beee976..4251199 100644
--- a/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
+++ b/src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/BufferedLineReader.kt
@@ -6,6 +6,8 @@ package com.github.doyaaaaaken.kotlincsv.client
 internal class BufferedLineReader(
     private val br: Reader
 ) {
+    private var isFirstLine: Boolean = true
+
     companion object {
         private val BOM = '\uFEFF'
     }
@@ -16,7 +18,8 @@ internal class BufferedLineReader(
             val c = br.read()
 
             if (c == -1) {
-                if (sb.isEmpty() || (sb.length == 1 && sb[0] == BOM)) {
+                if (sb.isEmpty() || (isFirstLine && sb.length == 1 && sb[0] == BOM)) {
+                    isFirstLine = false
                     return null
                 } else {
                     break
@@ -43,6 +46,7 @@ internal class BufferedLineReader(
                 break
             }
         } while (true)
+        isFirstLine = false
         return sb.toString()
     }
 
```